python/lsst/analysis/drp/gatherResourceUsage.py

```diff
@@ -0,0 +1,550 @@
+        for ref in metadata_refs:
+            metadata_refs_by_dataset_type[ref.datasetType].add(ref)
+        # Iterate over those groups, creating a configuration of
+        # this task and quanta for each one.
+        quanta_by_task_def = {}
+        for input_metadata_dataset_type, metadata_refs in metadata_refs_by_dataset_type.items():
+            if not input_metadata_dataset_type.name.endswith("_metadata"):
+                continue
+            input_task_label = input_metadata_dataset_type.name[:-len("_metadata")]
+            _LOG.info(
+                "Creating GatherResourceUsage quantum for %s with %d input datasets.",
+                input_task_label,
+                len(metadata_refs),
+            )
+            config = cls.ConfigClass()
+            config.dimensions = input_metadata_dataset_type.dimensions.names
+            config.connections.input_metadata = input_metadata_dataset_type.name
+            config.connections.output_table = f"{input_task_label}_resource_usage"
+            task_def = TaskDef(
+                taskName=get_full_type_name(cls),
+                taskClass=cls,
+                config=config,
+                label=f"{input_task_label}_gatherResourceUsage",
+            )
+            empty_dimensions = input_metadata_dataset_type.dimensions.universe.empty
+            output_table_dataset_type = DatasetType(
+                config.connections.output_table,
+                dimensions=empty_dimensions,
+                storageClass=GatherResourceUsageConnections.output_table.storageClass,
+            )
+            data_id = DataCoordinate.makeEmpty(universe=input_metadata_dataset_type.dimensions.universe)
+            outputs = {
+                output_table_dataset_type: [DatasetRef(output_table_dataset_type, data_id)],
+            }
+            if task_def.metadataDatasetName is not None:
+                output_metadata_dataset_type = DatasetType(
+                    task_def.metadataDatasetName,
+                    dimensions=empty_dimensions,
+                    storageClass="TaskMetadata",
+                )
+                outputs[output_metadata_dataset_type] = [DatasetRef(output_metadata_dataset_type, data_id)]
+            if task_def.logOutputDatasetName is not None:
+                log_dataset_type = DatasetType(
+                    task_def.logOutputDatasetName,
+                    dimensions=empty_dimensions,
+                    storageClass="ButlerLogRecords",
+                )
+                outputs[log_dataset_type] = [DatasetRef(log_dataset_type, data_id)]
+            quantum = Quantum(
+                taskName=task_def.taskName,
+                taskClass=task_def.taskClass,
+                dataId=data_id,
+                inputs={input_metadata_dataset_type: list(metadata_refs)},
+                outputs=outputs,
+            )
+            quanta_by_task_def[task_def] = {quantum}
+        return QuantumGraph(quanta_by_task_def)
```

Comment:
Do you think you want to check for cycles here, or subset to connected, and verify you have the expected number of independent chains (possibly not, I think you have just a bunch of independent quanta, but I wanted to raise it so it would be on your mind.)

Reply:
Should always be a bunch of independent quanta.